### PR TITLE
Allow reactions included in depletion chain to be configured

### PR DESCRIPTION
**Pull request by paulromano**
This pull request expands the list of (possible) transmutation reactions in the deplete/chain.py module. There is now a global dictionary, `REACTIONS`, that contains all the necessary information for each transmutation reaction. To actually choose which reactions should be included in a depletion chain, the `Chain.from_endf` method now has a `reactions` argument that takes an iterable of reaction names, e.g. `["(n,2n)", "(n,3n)", "(n,gamma)"]`. The default list of reactions is still the same as it was before.

I've been able to use these capabilities to generate a depletion chain using TENDL-2019 with 36 reactions present (for fusion applications). The way it works is roughly:

```python
from openmc.deplete import chain

chain.Chain.from_endf(..., reactions=chain.REACTIONS.keys())
```
A few other small fixes/changes:
- Made the reaction name for MT=34 consistent with others
- Fixed the reaction name for MT=180
- If a reaction product is missing during depletion chain generation, the `replace_missing` function is used to find a suitable replacement

### CODE REVIEW

File: openmc/deplete/chain.py [hunk 1]
```diff
@@ -256,6 +304,10 @@ def from_endf(cls, decay_files, fpy_files, neutron_files, progress=True):
             List of ENDF neutron-induced fission product yield sub-library files
         neutron_files : list of str or openmc.data.endf.Evaluation
             List of ENDF neutron reaction sub-library files
+        reactions : iterable of str, optional
+            Transmutation reactions to include in the depletion chain, e.g.,
+            `["(n,2n)", "(n,gamma)"]`. Note that fission is always included if
+            it is present.
```
Comment (Shimwell) [inline openmc/deplete/chain.py:310]: Perhaps something could be added to this doc string so users are more aware of that useful list of reactions in chain.Reactions.keys()

**Comment by Shimwell** [comment]
That looks most useful.

reactions using the same defaults as before but also allowing a much larger list to be added.

Not sure I would have spotted that useful list of reactions available in chain.REACTIONS.keys() without seeing your example and reading the source code. Wondering if that could be hinted at in the documentation or docstring.

File: openmc/deplete/chain.py (hunk 1 above)
Comment (paulromano) [reply openmc/deplete/chain.py]: Good suggestion @Shimwell. I've just pushed a commit that adds documentation of `REACTIONS` and references it here.

Comment (Shimwell) [reply openmc/deplete/chain.py]: thanks for that

**Comment by drewejohnson** [comment]
Based on the feedback by @shimwell https://github.com/openmc-dev/openmc/pull/1618#discussion_r462320718 and the updated documentation 3d35cd3, this looks good to go. Thanks @paulromano and @Shimwell!

**Pull request merged**